Kconfig.zephyr

```diff
@@ -167,12 +167,11 @@ config CMAKE_LINKER_GENERATOR
 
 endchoice
 
-config LLVM_USE_LD
-	bool "LLVM use ld linker"
+config LLVM_USE_LLD
```

Comment:
commit message states:
> This also changes CONFIG_LLVM_USE_LD to CONFIG_LLVM_USE_LLD.
> Since they are of opposite usage, it cannot utilize the usual
> deprecation process since that would use "select" to enable
> both, which cannot not be done in this case.

not understood ?
There is no `select LLVM_USE_LD`, there is only a `default y` in existing setting.

This can easily be changed into a choice, like this:
```
choice LLVM_LINKER
      prompt "LLVM Linker"
      depends on "${ZEPHYR_TOOLCHAIN_VARIANT}" = "llvm"
      default LLVM_USE_LD

config LLVM_USE_LD
      bool "LLVM use ld linker"
      help
        <existing help text>

config LLVM_USE_LLD
      bool "LLVM use lld linker"
      help
        <new help text>

endchoice
```
which will keep `ld` the default linker, and still provide a possibility to select `lld`

**note**: I have a feeling this code should actually be placed here instead:
https://github.com/zephyrproject-rtos/zephyr/tree/main/cmake/toolchain/llvm/Kconfig